ros2doctor/ros2doctor/api/__init__.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import platform
+
+import rosdistro
+
+
+def print_platform_info():
+    """Print out platform information."""
+    # platform info
+    print('System Information')
+    print('system   : ', platform.system())
+    print('Linux distribution   : ', platform.dist())
+    print('Mac OS version       : ', platform.mac_ver())
+    print('release  : ', platform.release())
+    print('processor: ', platform.processor())
+    print('\n')
+
+    # python info
+    print('Python')
+    print('version      : ', platform.python_version())
+    print('compiler     : ', platform.python_compiler())
+    print('build        : ', platform.python_build())
+    print('\n')
+
+
+def print_ros2_reqs():
+    """Print out ROS2 distribution info using `rosdistro`."""
+    distro_name = os.environ.get('ROS_DISTRO').lower()
+    u = rosdistro.get_index_url()
+    i = rosdistro.get_index(u)
+    distro_info = i.distributions.get(distro_name)
+    distro_data = rosdistro.get_distribution(i, distro_name).get_data()
+
+    print('ROS Information')
+    print('distribution name    : ', distro_name)
+    print('distribution type    : ', distro_info.get('distribution_type'))
+    print('distribution status  : ', distro_info.get('distribution_status'))
```

Comment:
It's possible someone set an environment variable `ROSDISTRO_INDEX_URL` to the non-v4 version of the index file, which doesn't have the type or status attributes. I think `rosdistro` will use that URL instead. What do these lines return if this check in `ros2 doctor` is invoked with?

```
ROSDISTRO_INDEX_URL https://raw.githubusercontent.com/ros/rosdistro/master/index.yaml ros2 doctor
```

Reply:
ROS Information
distribution name    :  dashing
distribution type    :  None
distribution status  :  None
release platforms    :  {'ubuntu': ['bionic']}

In this case,  status check is not invoked. I could add a warning regarding the two missing values if you think setting URL to non-v4 version presents significance in checks; otherwise, it seems ok to me since both distro_name and platform are still there.

Reply:
The output seems ok to me too. I was mostly wondering if `distro_info.get()` would raise an exception.

Reply:
If I used `distro_info['distribution_type']`, it would raise an exception. `.get()` just returns None if there's no such key and doesn't break anything, which is nice :)